From 3ceb4ef40e31c17c51e56418971a3ce1a6bcdf6e Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Thu, 2 Jul 2026 12:50:50 +0200
Subject: [PATCH] stages/bootc.install-to-fs: remount /dev/shm in addition to
 /dev/shm

In commit c1f5f50d8ce6340dae98fb2d3604dcfc65ce546e we introduced the
remounting of /dev. However, wit this /dev, podman, as started from
image-builder ends up with this error:

```
Error: failed to get new shm lock manager: failed to create 2048 locks in /libpod_lock: no such file or directory
```

This is due to the lack of /dev/shm, so mount it as well

Signed-off-by: Alexander Larsson <alexl@redhat.com>
---
 stages/org.osbuild.bootc.install-to-filesystem | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/stages/org.osbuild.bootc.install-to-filesystem b/stages/org.osbuild.bootc.install-to-filesystem
index 654711ca..2c67d914 100755
--- a/stages/org.osbuild.bootc.install-to-filesystem
+++ b/stages/org.osbuild.bootc.install-to-filesystem
@@ -77,10 +77,13 @@ def main(options, inputs, paths):  # pylint: disable=too-many-branches
         # to use the `composefs` backend based on the input container; we can't control
         # this so we should always be prepared.
         subprocess.check_call(['mount', '-t', 'devtmpfs', 'devtmpfs', '/dev/'])
+        os.makedirs('/dev/shm', exist_ok=True)
+        subprocess.check_call(['mount', '-t', 'tmpfs', 'tmpfs', '/dev/shm'])
 
         try:
             subprocess.run(pargs, env=env, check=True)
         finally:
+            subprocess.check_call(['umount', '/dev/shm'])
             subprocess.check_call(['umount', '/dev/'])
 
 
-- 
2.54.0

